errors.go

```diff
@@ -280,3 +280,39 @@ func Cause(err error) error {
 	}
 	return err
 }
+
+// Stack returns the earliest/deepest StackTrace attached to any of
+// the errors in the chain of Causes. If the error does not implement
+// Cause, the original error will be returned. If the error is nil,
+// nil will be returned without further investigation.
+func Stack(err error) StackTrace {
+	type causer interface {
+		Cause() error
+	}
+
+	type stackTracer interface {
+		StackTrace() StackTrace
+	}
```

Comment:
You shouldn't need to define these types here--these are already global types.

Reply:
Sorry, but I didn't find any public define of these types. Only private ones.

Reply:
This one is also private. Do you need a public one?

Reply:
I'm not sure about that. I don't know why the previous ones are all private.

Reply:
We want to keep the scope of the interfaces to the smallest set available. Firstly, because defining these interfaces scoped to the function that uses them makes it more clear what the interface is (it’s defined right above you).  Secondly, it also prevents us from exporting interface types unnecessarily (one cannot export something that is not even there) Finally, there’s considerations that we should be avoiding using interfaces as much as possible. Giving some overhead required to actually make use of them is a good way to deter rampant interface usage.

Basically, we should generally be avoiding defining package-global interfaces as much as possible.